clarity/src/vm/functions/arithmetic.rs

```diff
@@ -221,9 +221,44 @@ macro_rules! make_comparison_ops {
 macro_rules! make_arithmetic_ops {
     ($struct_name: ident, $type:ty) => {
         impl $struct_name {
+            fn bitwise_left_shift(x: $type, y: u32) -> InterpreterResult<Value> {
```

Comment:
Per my comment here [1], this should simply evaluate to 0 if `y > 128`.  So, the code should read:

```rust
fn bitwise_left_shift(x: $type, y: u32) -> InterpreterResult<Value> {
   if y > 128 {
      Self::make_value(0)
    }
    else {
      Self::make_value(x
         .checked_shl(y)
         // NOTE: error path should be unreachable
         .ok_or(RuntimeErrorType::ArithmeticOverflow)?)
    }
 }
 ```
Please continue to use `checked_shl` and `checked_shr`.

Reply:
Oops, missed this comment before pushing changes using `wrapped`-variants based on my discussion with @obycode.  Do you mind having a look at the new implementation, and seeing if it's acceptable before I go back to `checked`-variants?

Reply:
@obycode agrees with me: https://github.com/stacks-network/stacks-blockchain/pull/3382#issuecomment-1312499592

Reply:
Ah, I see you meant this: https://github.com/stacksgov/sips/pull/106#discussion_r1020840929.

Sure, `wrapping_shl` and `wrapping_shr` are fine in principle.

Reply:
Can these versions be removed?

Reply:
@obycode `<<` and `>>` panic.  Could change to checked, but they require that we return an error or pre-determined value.. 🤷 I don't know what's best.